### ISSUE TITLE
Loading `expl3-generic.tex` in LuaMetaTeX fails with `expl3.lua:421: table index is nil` since TeX Live 2024

### ISSUE THREAD

**Issue by Witiko** [latex3/latex3#1518]
Take the following MWE document `example.tex`:

```
\input expl3-generic
```

Compiling this document using the command `context example.tex` on TeX Live 2024 fails with the following output:

```
open source     > level 1, order 3, name './example.tex'
open source     > level 2, order 4, name 'expl3-generic.tex'
Package: expl3 2024-02-20 L3 programming layer (loader)

open source     > level 3, order 5, name 'expl3-code.tex'
Package: expl3 2024-02-20 L3 programming layer (code)

lua error       > lua error on line 154 in file expl3-code.tex:

token call, execute: ...cal/texlive/2024/texmf-dist/tex/latex/l3kernel/expl3.lua:421: table index is nil
stack traceback:
	...cal/texlive/2024/texmf-dist/tex/latex/l3kernel/expl3.lua:421: in main chunk
	[C]: in upvalue 'requiem'
	...live/2024/texmf-dist/tex/context/base/mkiv/l-sandbox.lua:180: in function <...live/2024/texmf-dist/tex/context/base/mkiv/l-sandbox.lua:165>
	(...tail calls...)
	[ctxlua]:1: in main chunk
<empty file>
mtx-context     | fatal error: return code: 1
```

Here is the corresponding line 421 in `expl3.lua`:

```
    [cmd'assign_mu_glue' or cmd'register_mu_glue'] = index_not_nil,
```

Compiling the same document using the same command on TeX Live 2023 or on TeX Live 2024 but using the command `context --luatex example.tex` succeeds. This indicates that since TeX Live 2024, `expl3-generic.tex` fails to load with LuaMetaTeX, whereas it succeeds with LuaTeX.

**Comment by josephwright** [comment]
I've tracked down this part: LMTX has renamed a few functions again. (Aside: we need to decide whether to keep the intermediate names or to drop them).

There's a second issue: I'm working on that too.

**Comment by josephwright** [comment]
Right, second issue is that LuaMetaTeX accepts `;` as part of `\numexpr` (seems to be undocumented) - we use that a lot as a delimiter, so need to change it.

**Comment by josephwright** [comment]
OK, I'll check: my feeling is this is by-design but you never know.

**Comment by josephwright** [comment]
Seems that support for `;` is by-design in LuaMetaTeX `\numexpr`. Changing the `expl3` code to use a different delimiter is going to be non-trivial: I will look at this but it won't be for a little while.

**Comment by josephwright** [comment]
I'm still wondering if there is any chance of getting Hans to look again at his `\numexpr` change ...

**Comment by Witiko** [comment]
Did any exchange with Hans take place? Should I nag him about it?

**Comment by josephwright** [comment]
> Did any exchange with Hans take place? Should I nag him about it?

Not yet, but looking back and with the newer primitive `\numexpression` available, if I can convince him about `\numexpr` it would be easiest!

**Comment by josephwright** [comment]
We've thought about this, and it's really an engine thing - we support `\numexpr` with the e-TeX syntax, not with the `;` extension. So this needs to be raised with the LuaMetaTeX people.

**Comment by Witiko** [comment]
Got it! I will go and bother Hans instead.

**Comment by josephwright** [comment]
We should probably add a test for LuaMetaTeX - I'll sort that once the PR is merged.

**Comment by josephwright** [comment]
Now all working - closing as completed -  I have also added (very light) testing so we will know if `expl3` fails to load entirely in future.
